src/main/java/graphql/execution/Async.java

```diff
@@ -135,29 +135,16 @@ public CompletableFuture<List<T>> await() {
         CompletableFuture<U> apply(T input, int index, List<U> previousResults);
     }
 
-    public static <U> CompletableFuture<List<U>> each(List<CompletableFuture<U>> futures) {
-        CompletableFuture<List<U>> overallResult = new CompletableFuture<>();
-
-        @SuppressWarnings("unchecked")
-        CompletableFuture<U>[] arrayOfFutures = futures.toArray(new CompletableFuture[0]);
-        CompletableFuture
-                .allOf(arrayOfFutures)
-                .whenComplete((ignored, exception) -> {
-                    if (exception != null) {
-                        overallResult.completeExceptionally(exception);
-                        return;
-                    }
-                    List<U> results = new ArrayList<>(arrayOfFutures.length);
-                    for (CompletableFuture<U> future : arrayOfFutures) {
-                        results.add(future.join());
-                    }
-                    overallResult.complete(results);
-                });
-        return overallResult;
+    public static <U> CompletableFuture<List<U>> each(List<CompletableFuture<U>> futures) { // TODO: used only in tests now
+        CombinedBuilder<U> overall = ofExpectedSize(futures.size());
```

Comment:
You could also created a method that produced a build of the right size in one call

```
CombinedBuilder<U> overall = fromList(futures);
```

There is no need to add them all one by one outside that method.

You can do the same code as here inside the `fromList` method say

Reply:
I tried that but I stopped for 2 reasons:
- that will defeat the idea of the CombinedBuilder (i.e. avoid List -> array allocation)
-  this method is used only for tests now... and since it is @Internal, we should be fine no?